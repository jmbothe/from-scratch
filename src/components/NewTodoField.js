import React, { Fragment } from 'react';

const NewTodoField = ({ meta: { error } }) => 
  <Fragment>
    <input
      id="new-todo-input"
      type="text"
      placeholder="Add a new todo..."
    />
    <p>{ error }</p>
  </Fragment>
 
export default NewTodoField;