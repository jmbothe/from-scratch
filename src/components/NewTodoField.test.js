import React from 'react';
import { shallow } from 'enzyme';

import NewTodoField from './NewTodoField';

describe('<NewTodoField />', () => {
  let props = { meta: { error: null } }
  let newTodoField = shallow(<NewTodoField { ...props } />);

  it('renders correctly', () => {
    expect(newTodoField).toMatchSnapshot();
  })

  it('renders a text input element', () => {
    expect(newTodoField.find('input').props().type).toEqual('text');
  })

  describe('when input field is empty', () => {
    it('displays a placeholder', () => {
      expect(newTodoField.find('input').props().placeholder.text()).toEqual('Add a new todo...');
    })
  })

  describe('when not in error state', () => {
    it('does not display an error message', () => {
      expect(newTodoField.children()[1].text()).toEqual('');
    })
  })

  describe('when in error state', () => {
    let props = { meta: { error: 'Please enter a todo before submitting' } }
    let newTodoField = shallow(<NewTodoField { ...props } />);
    it('displays an error message', () => {
      expect(newTodoField.children()[1].text()).toEqual('Please enter a todo before submitting');
    })
  })

})