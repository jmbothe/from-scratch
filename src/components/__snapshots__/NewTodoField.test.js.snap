// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewTodoField /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewTodoField
    meta={
      Object {
        "error": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          id="new-todo-input"
          placeholder="Add a new todo..."
          type="text"
        />,
        <p />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "new-todo-input",
          "placeholder": "Add a new todo...",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            id="new-todo-input"
            placeholder="Add a new todo..."
            type="text"
          />,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "new-todo-input",
            "placeholder": "Add a new todo...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
